mlflow/models/evaluation/base.py

```diff
@@ -131,8 +136,8 @@ def make_metric(
 
             def eval_fn(
                 eval_df: Union[pandas.Dataframe, pyspark.sql.DataFrame],
-                builtin_metrics: Dict[str, float],
-            ) -> float:
+                metrics: Dict[str, MetricValue],
```

Comment:
@annzhang-db How do we intend to have backwards compatibility with `builtin_metrics`? Is the backcompat logic coming in a follow-up PR?

Reply:
Yes, the logic for updating mlflow.evaluate() will be coming in a [follow-up PR](https://github.com/mlflow/mlflow/pull/9440/files) but it hasn't been modified for this definition of eval_fn yet

Reply:
[Here](https://github.com/mlflow/mlflow/pull/9440/files#diff-1b3ab8e1152a42ab0bd3d4e01c7621b37a95aa02fff19642a5abf23f75b6e688R483)'s the code for accepting float output from eval_fn

Reply:
Awesome - sounds good!